packages/sdk/src/CoreContracts.ts

```diff
@@ -29,7 +29,8 @@ export class CoreContracts<T extends NomadContext> extends Contracts<
     this.conf = conf;
   }
 
-  getReplica(domain: string): core.Replica | undefined {
+  getReplica(nameOrDomain: string | number): core.Replica | undefined {
+    const domain = this.context.resolveDomain(nameOrDomain);
     if (!this.connection) throw new NoProviderError(this.context, domain);
     const replica = this.conf.replicas[domain];
```

Comment:
why is this not a type error?

`domain` is a `number`
`this.conf.replicas` is a `Record<string, _>`

why can we index it with a number? this is a bug. Need to use `resolveDomainName` on L33